WP-1185 fix SDK using proxy

pengyuc-bitgo opened the pull request:
The `superagent-proxy` was removed earlier in this [PR](https://github.com/BitGo/BitGoJS/pull/3921).
And we have to use the proxy-agent now. But that change missed some old usage of the `superagent-proxy`.

This PR also changed the proxy-agent version to 5.0. This is the version that was used under the hood before. The new version requires setting proxy url in the environment variable. It does not quite fit how we use the proxy-agent currently. If we want to upgrade to the latest proxy-agent, it should be done in a separate PR.

This change is tested with the new example codes and local proxy server. Verified that it is using the proxy by shutting down the proxy and saw the connection failure error.

- [x] Bug fix (non-breaking change which fixes an issue)

Ticket: WP-1185

pengyuc-bitgo committed:
fix(sdk-api): fix proxy-agent usage

We removed the superagent-proxy earlier to use proxy-agent.
The change missed some old usage.

Change the proxy-agent version to 5.0. This is the version
that was used under the hood. The new version requires setting
proxy url in the environment variable. It does not quite fit how
we use the proxy-agent currently.

Ticket: WP-1185

zahin-mohammad commented on examples/ts/proxy/server.ts:
```diff
@@ -1,29 +1,8 @@
+import { AddressInfo } from 'net';
+const proxy = createProxy();
 
-// This *must* match the environment you are using in the create-wallet script
-const bitgoApi = 'https://app.bitgo-test.com';
-// TODO: replace with your access token
-const secretAccessToken = 'yourAccessToken';
```
Why are we changing this example? The goal of this example is to show an example of the server injecting the access token opposed to needing it in the frontend.

pengyuc-bitgo replied:
I have roll-back the changes to this example and added another example for http proxy usage.
Got confused because of the name. The use case are actually different.

zahin-mohammad reviewed (commented):
Changes lgtm, but I think the examples for create-wallet/server should stay the same. Feel free to sync with @alebusse to see why its having issues (we have clients using similar setups so we should ensure it still works)

pengyuc-bitgo committed:
test(sdk-api): a new example to show and test http proxy usage

Ticket: WP-1185

pengyuc-bitgo commented:
> Changes lgtm, but I think the examples for create-wallet/server should stay the same. Feel free to sync with @alebusse to see why its having issues (we have clients using similar setups so we should ensure it still works)

Got confused by the example's name. Added another example for http proxy. Also updated the README.md.
